### PR TITLE
Remove Fedora 21 from RPM build as it is EOL

### PR DESCRIPTION
**Pull request by thaJeztah**
Fedora 21 is EOL'd as of December 1st, 2015, so we should not publish RPMs for Docker 1.10

Announcement:
https://lists.fedoraproject.org/pipermail/announce/2015-November/003296.html

> Fedora 21 will reach end of life on 2015-12-01, and no further updates
> will be pushed out after that time. Additionally, with the recent
> release of Fedora 23, no new packages will be added to the Fedora 21
> collection.
> 
> Please see https://fedoraproject.org/wiki/DNF_system_upgrade for more
> information on upgrading from Fedora 21 to a newer release.

### CODE REVIEW

File: contrib/builder/rpm/generate.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ for version in "${versions[@]}"; do
 	suite="${version##*-}"
 	from="${distro}:${suite}"
 	installer=yum
-	if [[ "$distro" == "fedora" ]] && [[ "$suite" -ge "22" ]]; then
+	if [[ "$distro" == "fedora" ]]; then
```
Comment (thaJeztah) [inline contrib/builder/rpm/generate.sh]: not sure if all current CentOS and RHEL releases also have dnf installed by default now

**Comment by thaJeztah** [comment]
ping @jfrazelle @tianon

File: contrib/builder/rpm/generate.sh (hunk 1 above)
Comment (jfrazelle) [inline contrib/builder/rpm/generate.sh]: ya centos does not but this LGTM awesome!

**Comment by jfrazelle** [comment]
LGTM

**Comment by thaJeztah** [comment]
Also see this PR https://github.com/docker/docker/issues/18790, which can be merged if this accepted :)

File: docs/installation/fedora.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ weight=-3
 
 # Fedora
 
-Docker is supported Fedora version 21 and 22. This page instructs you to install
+Docker is supported Fedora version 22 and 23. This page instructs you to install
```
Comment (thaJeztah) [inline docs/installation/fedora.md]: oh, see that this misses "on". let me fix that as well

File: contrib/builder/rpm/generate.sh (hunk 1 above)
Comment (thaJeztah) [inline contrib/builder/rpm/generate.sh]: yeah, we can tweak later, just was wondering when I saw this, haha

**Comment by runcom** [comment]
LGTM

**Pull request merged**